### ISSUE TITLE
URL-Encoded plus signs are re-encoded as whitespaces

### ISSUE THREAD

**Issue by TimZolleis** [steven-tey/dub#153]
**TLDR: Dub replaces URL-Encoded "plus" signs (%2B) by whitespaces (%20)**

I expected some really weird behaviour on links that use URL-Encoded plus signs (in this specific use-case with Apple maps guide links) - which breaks them effectively. However, im not quite sure where this issue originates from on dub.sh, as I was **not able to reproduce this issue locally**

#### Example:

```
https://guides.apple.com/?ug=CglEVUIgR3VpZGUSDgjZMhDEo%2BGA%2BZKqpJUBEg4I2TIQw7y33%2B%2B6ifL%2BARIOCNkyEJC988jqgIrQjQESDgjZMhCB%2B7XSiPTwrfUBEg4I2TIQ5J25xZOynPDxARINCNkyENuVr4POz8aMcBIOCMI7EK36pfjQuerJ0gESDQjCOxDSuurnjM6T7mASDQjCOxD3vr%2F%2Fkq%2FLqUwSDQjCOxCg9cK%2BjeOhnS4%3D
```
gets replaced by 

```
https://guides.apple.com/?ug=CglEVUIgR3VpZGUSDgjZMhDEo%20GA%20ZKqpJUBEg4I2TIQw7y33%20%206ifL%20ARIOCNkyEJC988jqgIrQjQESDgjZMhCB%207XSiPTwrfUBEg4I2TIQ5J25xZOynPDxARINCNkyENuVr4POz8aMcBIOCMI7EK36pfjQuerJ0gESDQjCOxDSuurnjM6T7mASDQjCOxD3vr%2F%2Fkq%2FLqUwSDQjCOxCg9cK%20jeOhnS4%3D
```
Which effectively replaced all "%2B" with "%20"

Since the links are stored correctly in the DB and thus work when copying them, it must be with the middleware somehow? Maybe the "URL" in the NextRequest.redirect() tampering with the encoding?

As this does not happen locally, is this maybe some vercel-specific behaviour?
Attached local package.json for dependency checking...

Cheers!
[package.json.zip](https://github.com/steven-tey/dub/files/10806572/package.json.zip)

**Comment by steven-tey** [comment]
I [fixed](https://github.com/steven-tey/dub/commit/9ffc677c56b18bc10c682f6ae494d0a533c189d7) this issue! Turns out it was a rookie mistake of not encoding the URL properly when it's created 😅 

Sorry for the inconvenience!

**Comment by TimZolleis** [comment]
Tysm! No problem at all!

Weird that i didn't experience when trying it out locally, maybe some wrong Redis setup...

Thanks for the quick fix!
